src/test/fixtures/profiles/make-profile.js

```diff
@@ -144,18 +144,22 @@ export function getEmptyThread(overrides: ?Object): Thread {
  *
  * ```
  * const { profile } = getProfileFromTextSamples(`
- *   A       A        A     A
- *   B.js    B.js     F     F
- *   C.js    C.js     G     G
- *   D       D              H
- *   E       E
+ *   A          A         A             A
+ *   B.js       B.js      F [Graphics]  F
+ *   C.js       C.js      G             G
+ *   D          D                       H [DOM]
+ *   E [Other]  E [Other]
  * `);
  * ```
  *
  * The function names are aligned vertically on the left. This would produce 4 samples
  * with the stacks based off of those functions listed, with A being the root. Single
  * spaces within a column are permitted, surrounding whitespace is trimmed.
  *
+ * Functions ending in "js" are marked as JS functions in the funcTable's isJS
+ * column. Functions that have a category name in square brackets after them
+ * are annotated with that category in the frameTable's category column.
+ *
```

Comment:
A quick note because I changed this function in https://github.com/devtools-html/perf.html/pull/1041/files#diff-f74cb1c027c114cfd23055a90af53ee5 to be able to specify optimizations after all. I used a similar solution to yours by using backets, but still sticked to the function name, and with a prefix, eg: `Cjs[jit:ion]`.